src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.Integer.cs

```diff
@@ -26,6 +27,13 @@ public void WriteInt64(long value)
             {
                 WriteUnsignedInteger(CborMajorType.UnsignedInteger, (ulong)value);
             }
+
+            DecrementRemainingItemCount();
+        }
+
+        public void WriteTag(CborTag tag)
+        {
+            WriteUnsignedInteger(CborMajorType.Tag, (ulong)tag);
```

Comment:
Can a tag have no payload?  If not, doesn't it need to set an expectation of further data?

I don't seem to see a test like

```C#
CborWriter writer = new CborWriter();
writer.WriteTag(3);
// Presumably this should throw for incomplete data...
writer.ToArray();
```

Reply:
Yes, this is expected to fail. I'll add a test.

Reply:
Ooh, does this end up causing failure because it doesn't decrement the count remaining? If so, perhaps add a comment for that.

But I'm a little concerned that it'll currently not fail in an indefinite array/map.

```C#
writer.WriteStartMapIndefiniteLength();
writer.WriteTag(something);
// count is still %2 because we didn't finish the value
writer.WriteEndMap();
writer.ToArray();
```

Reply:
Good point, I've added a flag for tracking cbor contexts, it should catch that case and provide more meaningful error messages in general.